pkg/crc/cluster/cluster.go

```diff
@@ -119,20 +119,56 @@ func UpdateClusterID(ocConfig oc.Config) error {
 	return nil
 }
 
-func AddProxyConfigToCluster(ocConfig oc.Config, proxy *network.ProxyConfig) error {
+func AddProxyConfigToCluster(sshRunner *ssh.Runner, ocConfig oc.Config, proxy *network.ProxyConfig) error {
+	trustedCAName := "user-ca-bundle"
 	cmdArgs := []string{"patch", "proxy", "cluster", "-p",
 		fmt.Sprintf(`'{"spec":{"httpProxy":"%s", "httpsProxy":"%s", "noProxy":"%s"}}'`, proxy.HTTPProxy, proxy.HTTPSProxy, proxy.GetNoProxyString()),
 		"-n", "openshift-config", "--type", "merge"}
 
 	if err := WaitForOpenshiftResource(ocConfig, "proxy"); err != nil {
 		return err
 	}
+	if proxy.ProxyCACert != "" {
+		logging.Debug("Adding proxy CA cert to cluster")
+		if err := addProxyCACertToCluster(sshRunner, ocConfig, proxy, trustedCAName); err != nil {
+			return err
+		}
+		cmdArgs = []string{"patch", "proxy", "cluster", "-p",
+			fmt.Sprintf(`'{"spec":{"httpProxy":"%s", "httpsProxy":"%s", "noProxy":"%s", "trustedCA":{"name":"%s"}}}'`,
+				proxy.HTTPProxy, proxy.HTTPSProxy, proxy.GetNoProxyString(), trustedCAName),
+			"-n", "openshift-config", "--type", "merge"}
+	}
 	if _, stderr, err := ocConfig.RunOcCommand(cmdArgs...); err != nil {
 		return fmt.Errorf("Failed to add proxy details %v: %s", err, stderr)
 	}
 	return nil
 }
 
+func addProxyCACertToCluster(sshRunner *ssh.Runner, ocConfig oc.Config, proxy *network.ProxyConfig, trustedCAName string) error {
+	proxyCABundleTemplate := `{
+  "apiVersion": "v1",
+  "data": {
+    "ca-bundle.crt": "%s"
+  },
+  "kind": "ConfigMap",
+  "metadata": {
+    "name": "%s",
+    "namespace": "openshift-config"
+  }
+}
+`
+	p := fmt.Sprintf(proxyCABundleTemplate, strings.ReplaceAll(proxy.ProxyCACert, "\n", `\n`), trustedCAName)
```

Comment:
slightly confused here? is this replacing actual carriage returns with the caracters `\n` ? or is this doing something else?

Reply:
@cfergeau yes, I am adding same as comment.

Reply:
But all line endings were removed by `getProxyCAData`, so is there anything left to replace?

Reply:
@cfergeau not all, it only remove the end of file line not intermediate one.